api/v1alpha1/dnsrecord_types.go

```diff
@@ -101,31 +106,46 @@ const (
 	DefaultGeo string = "default"
 )
 
-// GetRootDomain returns the shortest domain that is shared across all spec.Endpoints dns names.
-// Validates that all endpoints share an equal root domain and returns an error if they don't.
-func (s *DNSRecord) GetRootDomain() (string, error) {
-	domain := ""
-	dnsNames := []string{}
-	for idx := range s.Spec.Endpoints {
-		dnsNames = append(dnsNames, s.Spec.Endpoints[idx].DNSName)
+const wildcardPrefix = "*."
+
+func (s *DNSRecord) isWildCardRoot() bool {
+	if s.Spec.RootHost != nil {
+		return strings.HasPrefix(*s.Spec.RootHost, wildcardPrefix)
 	}
-	for idx := range dnsNames {
-		if domain == "" || len(domain) > len(dnsNames[idx]) {
-			domain = dnsNames[idx]
+	return false
+}
+
+func (s *DNSRecord) Validate() error {
+	if s.Spec.RootHost != nil {
+		root := *s.Spec.RootHost
+		if len(strings.Split(root, ".")) <= 1 {
+			return fmt.Errorf("invalid domain format no tld discovered")
+		}
+		if len(s.Spec.Endpoints) == 0 {
+			return fmt.Errorf("no endpoints defined for DNSRecord. Nothing to do.")
 		}
-	}
 
-	if domain == "" {
-		return "", fmt.Errorf("unable to determine root domain from %v", dnsNames)
-	}
+		if s.isWildCardRoot() {
+			root = strings.Replace(root, wildcardPrefix, "", 1)
+		}
```

Comment:
You could arguably just always[ cut the prefix](
https://pkg.go.dev/strings#CutPrefix) all the time and remove `isWildCardRoot()` since you don't really care if it's found or not:

```
root, _ := strings.CutPrefix(root, wildcardPrefix)
```